drivers/serial/Kconfig

```diff
@@ -77,6 +77,27 @@ config UART_DRV_CMD
 
 	  Says no if not sure.
 
+config UART_PIPE
+	bool "Enable pipe UART driver"
+	select UART_INTERRUPT_DRIVEN
+	help
+	  Enable pipe UART driver. This driver allows application to communicate
+	  over UART with custom defined protocol. Driver doesn't inspect received
+	  data (as contrary to console UART driver) and all aspects of received
+	  protocol data are handled by application provided callback.
+
+# Workaround for not being able to have commas in macro arguments
+DT_CHOSEN_Z_UART_PIPE := zephyr,uart-pipe
+
+config UART_PIPE_ON_DEV_NAME
```

Comment:
Since you are touching it, please remove this UART_PIPE_ON_DEV_NAME, and replace 
[uart_pipe_dev = device_get_binding(CONFIG_UART_PIPE_ON_DEV_NAME);](https://github.com/zephyrproject-rtos/zephyr/pull/38204/commits/11463110231d6236733a57009cabddbdee5acde0#diff-da50ff9ee6e7b7e391b40c21414050398f503894ddd06dbd05c8bf5441b297a1R103) with DEVICE_DT_GET(DT_CHOSEN(uart_pipe));